File: frontend/src/types/branchDashboard.ts
export interface DashboardUser {
  name: string
  branch_name: string
  role: string
}

export interface DashboardShift {
  is_open: boolean
  session_id?: number | null
  opened_at?: string | null
  duration_minutes?: number | null
}

export interface TopProduct {
  name: string
  units: string
}

export interface DashboardToday {
  sales_total: string
  sales_count: number
  avg_ticket: string
  returns_total: string
  returns_count: number
  goal?: string | null
  goal_progress_pct?: number | null
  payment_methods?: Record<string, string> | null
  top_products?: TopProduct[] | null
}

export type AlertKind = 'low_stock' | 'no_branch_price' | 'quote_expiring' | 'cash_variance'

export interface DashboardAlert {
  kind: AlertKind
  count?: number | null
  amount?: string | null
  deeplink: string
}

export interface BranchDashboard {
  user: DashboardUser
  shift: DashboardShift
  today: DashboardToday
  alerts: DashboardAlert[]
  closing_visible: boolean
}

File: app/schemas/branch_dashboard.py
from datetime import datetime
from decimal import Decimal
from typing import Dict, List, Literal, Optional
from pydantic import BaseModel, ConfigDict


class DashboardUser(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    name: str
    branch_name: str
    role: str = ""


class DashboardShift(BaseModel):
    is_open: bool
    session_id: Optional[int] = None
    opened_at: Optional[datetime] = None
    duration_minutes: Optional[int] = None


class TopProduct(BaseModel):
    name: str
    units: Decimal


class DashboardToday(BaseModel):
    sales_total: Decimal
    sales_count: int
    avg_ticket: Decimal
    returns_total: Decimal
    returns_count: int
    goal: Optional[Decimal] = None
    goal_progress_pct: Optional[float] = None
    payment_methods: Optional[Dict[str, Decimal]] = None
    top_products: Optional[List[TopProduct]] = None


AlertKind = Literal["low_stock", "no_branch_price", "quote_expiring", "cash_variance"]


class DashboardAlert(BaseModel):
    kind: AlertKind
    count: Optional[int] = None      # used for low_stock, no_branch_price, quote_expiring
    amount: Optional[Decimal] = None # used for cash_variance
    deeplink: str


class BranchDashboardRead(BaseModel):
    user: DashboardUser
    shift: DashboardShift
    today: DashboardToday
    alerts: List[DashboardAlert]
    closing_visible: bool
